services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableManagerSupportImpl.java

```diff
@@ -756,4 +756,17 @@ public ActionsRequiredDao getActionsRequiredDao(IdAndVersion idAndVersion) {
 		return new ActionsRequiredDao(tableConnectionFactory.getConnection(idAndVersion).getConnection());
 	}
 
+	@Override
+	public void deleteTableStatus(IdAndVersion idAndVersion) {
+		ValidateArgument.required(idAndVersion, "idAndVersion");
+		ValidateArgument.required(idAndVersion.getId(), "idAndVersion.getId");
+
+		if (idAndVersion.getVersion().isEmpty() || idAndVersion.getVersion().get().equals(-1L)) {

```

Comment:
we need to depend on the isEmtpy() only.
Can we change the IdAndVersion to set version to null when -1